cluster/images/etcd/Makefile

```diff
@@ -26,12 +26,17 @@
 # Except from etcd-$(tag) and etcdctl-$(tag) binaries, we also
 # need etcd and etcdctl binaries for backward compatibility reasons.
 # That binary will be set to the last tag from $(TAGS).
-TAGS?=2.2.1 2.3.7 3.0.17 3.1.11
-REGISTRY_TAG?=3.1.11
+TAGS?=2.2.1 2.3.7 3.0.17 3.1.12
+REGISTRY_TAG?=3.1.12
 # ROLLBACK_REGISTRY_TAG specified the tag that REGISTRY_TAG may be rolled back to.
-ROLLBACK_REGISTRY_TAG?=3.1.11
+ROLLBACK_REGISTRY_TAG?=3.1.12
 ARCH?=amd64
+# Image should be pulled from k8s.gcr.io, which will auto-detect
+# region (us, eu, asia, ...) and pull from the closest.
 REGISTRY?=k8s.gcr.io
+# Images should be pushed to staging-k8s.gcr.io, which is aliased to gcr.io/google-containers.
+# But currently that does not work, so we must instead push to gcr.io/google-containers directly.
+PUSH_REGISTRY?=gcr.io/google-containers
```

Comment:
Pushing directly to staging-k8s.gcr.io did not work for me although from https://github.com/kubernetes/kubernetes/pull/57824 it looks like it is suppose to. Anyone know why?

cc @thockin

Reply:
Could you tell what the error is? Maybe it's related to the one I was seeing yesterday - https://github.com/kubernetes/kubernetes/pull/60891#discussion_r173246510

Reply:
I either get this (for layer pushes):
```
The push refers to a repository [staging-k8s.gcr.io/etcd]
21172253818c: Retrying in 5 seconds 
6689586d0bd6: Retrying in 4 seconds
```

Or, this (if layers already exist, but something needs to be tagged):

```
denied: Permission 'storage.objects.create' denied for bucket 'artifacts.google-containers.appspot.com'
```

Happens using both `docker <cmd>` and `gcloud docker -- <cmd>` works works if i instead do the same against `gcr.io/google-containers`.  Permissions issue?

Reply:
Did you try `gcloud auth login` and authenticate yourself as k8s.production.user@?

Reply:
also, make sure you have `gcloud` 185.0.0 or newer and run `gcloud beta auth configure-docker` - this should allow `docker push` to work.

Reply:
That fixed the problem. Thanks @ixdy!  I'll update this PR.